Mathlib/Data/Fin/Parity.lean

```diff
@@ -94,4 +94,15 @@ lemma not_even_iff_odd_of_even {n : ℕ} [NeZero n] (hn : Even n) {k : Fin n} :
   rw [even_iff_of_even hn, odd_iff_of_even hn]
   exact Nat.not_even_iff_odd
 
+lemma odd_add_one_iff_even {n : ℕ} [NeZero n] {k : Fin n} : Odd (k + 1) ↔ Even k := by
+  refine ⟨fun ⟨k, hk⟩ ↦ ?_, Even.add_one⟩
+  exact add_right_cancel hk ▸ even_two_mul k
+
+lemma even_add_one_iff_odd {n : ℕ} [NeZero n] {k : Fin n} : Even (k + 1) ↔ Odd k := by
+  refine ⟨fun ⟨k, hk⟩ ↦ ?_, Odd.add_one⟩
+  have h := congrFun (congrArg HSub.hSub hk) 1
+  rw [add_sub_cancel_right] at h
+  rw [h]
```

Comment:
```suggestion
  rw [eq_sub_iff_add_eq.mpr hk]
```

Reply:
I did something similar but with the same style that in `odd_add_one_iff_even`, is okay?

Reply:
Up to you I guess, I'm a newbie too.

Reply:
Okey, thanks